src/ucs/sys/sys.c

```diff
@@ -1079,3 +1079,30 @@ void ucs_sys_free(void *ptr, size_t length)
         }
     }
 }
+
+char* ucs_make_affinity_str(const cpu_set_t *cpuset, char **p_str)
+{
+    int length = sizeof(*cpuset) * 8;
+    char *str;
+    int i;
+
+    while (!CPU_ISSET(length - 1, cpuset)) {
+        length--;
+    }
+
+    length = ucs_align_up(length, 32);
+    str = ucs_malloc(length + 1, "affinity_str");
+    if (str == NULL) {
+        goto out;
+    }
+
+    for (i = 0; i < length; i++) {
+        str[i] = CPU_ISSET(i, cpuset) ? '*' : '.';
```

Comment:
maybe write CPU ranges or masks?

Reply:
great space for fantasy, started for most naive approach